### DISCUSSION TITLE
Saving histograms to disk

### DISCUSSION THREAD

**Discussion by dguest** [scikit-hep/coffea#870] [Q&A] [answered]
Is there a recommended way to save histograms to disk? I'm not actually interested in using coffea (for the moment) but rather in writing some histograms in a compatible format from some much uglier code. If some schema exists it would be great.

**Comment by andrzejnovak** [comment, marked as answer]
It's an ongoing topic https://github.com/scikit-hep/boost-histogram/discussions/726. In practice, most of us tend to write them out via uproot to ROOT files for interoperability or pickle them if not for long-term storage.

**Reply by dguest** [reply]
Nice, it sounds like I'm using HDF5 or json!

**Reply by dguest** [reply]
I'll comment there when I manage to scrape something together.

**Comment by matthewfeickert** [comment]
Relevant to this discussion: @henryiii posted this in the IRIS-HEP Slack today following up discussions had at [PyHEP.dev 2023](https://indico.cern.ch/event/1293313/)

> Followup for Histogram serialization: A first draft with schema is at https://uhi--105.org.readthedocs.build/en/105/serialization.html and https://github.com/scikit-hep/uhi/pull/105
